### PR TITLE
[#11186] Change gRPC UNARY response based on header value

### PR DESCRIPTION
**Pull request by donghun-cho**

### CODE REVIEW

File: grpc/src/main/java/com/navercorp/pinpoint/grpc/Header.java
Comment (feelform) [inline grpc/src/main/java/com/navercorp/pinpoint/grpc/Header.java:1]: Good equals and hashCode implemetations

File: collector/src/main/java/com/navercorp/pinpoint/collector/receiver/grpc/service/SimpleRequestHandlerAdaptor.java [hunk 1]
```diff
@@ -49,7 +50,7 @@ public SimpleRequestHandlerAdaptor(String name, DispatchHandler<REQ, RES> dispat
     public void request(Message<? extends REQ> message, StreamObserver<? extends RES> responseObserver) {
         try {
             final ServerRequest<? extends REQ> request = serverRequestFactory.newServerRequest(message);
-            final ServerResponse<? extends RES> response = new GrpcServerResponse<>(responseObserver);
+            final ServerResponse<? extends RES> response = new GrpcServerResponse<>(responseObserver, ServerContext.getAgentInfo().isRetryFriendlyResponse());
```
Comment (feelform) [inline collector/src/main/java/com/navercorp/pinpoint/collector/receiver/grpc/service/SimpleRequestHandlerAdaptor.java]: Could you check `create new GrpcServerResponse type for the RetryFriendlyResponse`?

File: collector/src/main/java/com/navercorp/pinpoint/collector/receiver/grpc/GrpcServerResponse.java [hunk 2]
```diff
@@ -26,15 +28,28 @@
...
+    public GrpcServerResponse(StreamObserver<T> responseObserver, boolean retryFriendlyResponse) {
         this.responseObserver = Objects.requireNonNull(responseObserver, "responseObserver");
+        this.retryFriendlyResponse = retryFriendlyResponse;
     }
 
     @Override
     public void write(final T message) {
         Objects.requireNonNull(message, "message");
-
+        if (retryFriendlyResponse) {
```
Comment (feelform) [inline collector/src/main/java/com/navercorp/pinpoint/collector/receiver/grpc/GrpcServerResponse.java]: Could you check create new GrpcServerResponse type for the RetryFriendlyResponse?

**Review by feelform — COMMENTED** [review]
Could you check about the new GrpcServerResponse type for the RetryFriendlyResponse?

File: collector/src/main/java/com/navercorp/pinpoint/collector/receiver/grpc/service/SimpleRequestHandlerAdaptor.java (hunk 1 above)
Comment (donghun-cho) [reply collector/src/main/java/com/navercorp/pinpoint/collector/receiver/grpc/service/SimpleRequestHandlerAdaptor.java]: fixed with new GrpcRetryFriendlyServerResponse class

**Comment by feelform** [comment]
LGTM

**Pull request merged**